src/main/java/org/jabref/gui/fieldeditors/JournalInfoOptInDialogHelper.java

```diff
@@ -0,0 +1,29 @@
+package org.jabref.gui.fieldeditors;
+
+import org.jabref.gui.DialogService;
+import org.jabref.logic.journals.JournalInformationPreferences;
+import org.jabref.logic.l10n.Localization;
+
+public class JournalInfoOptInDialogHelper {
+
+    /**
+     * Using the journal information data fetcher service needs to be opt-in for GDPR compliance.
+     */
+    public static boolean isJournalInfoEnabled(DialogService dialogService, JournalInformationPreferences preferences) {
+        if (preferences.isJournalInfoEnabled()) {
+            return true;
+        }
+
+        if (preferences.isJournalInfoOptOut()) {
+            return false;
+        }
```

Comment:
We had a looong discussion on two settings for online services at https://github.com/koppor/jabref/issues/566.

I think, the summary is at https://github.com/JabRef/jabref/pull/9801#issuecomment-1524050565

We opted for an enum `FIRST_START, ENABLED, DISABLED`. You can introduce that generally as it will be used here and at Grobid (later by someone ^^).